ui/imports/shared/views/AssetsView.qml

```diff
@@ -62,42 +69,78 @@ ColumnLayout {
+            id: assetsWithFilteredBalances
+            sourceModel: root.assets
+            submodelRoleName: "balances"
+            delegateModel: SortFilterProxyModel {
+                sourceModel: submodel
+                filters: ExpressionFilter {
+                    expression: {
+                        root.networkFilters
+                        root.addressFilters
+                        return root.networkFilters.split(":").includes(chainId+"") &&
+                                (!!root.addressFilters ? root.addressFilters.toUpperCase() === account.toUpperCase() : true)
+                    }
+                }
+            }
+        }
 
-        sourceModel: root.assets
+        function getTotalBalance(balances, decimals) {
+            let totalBalance = 0
+            for(let i=0; i<balances.count; i++) {
+                totalBalance+=SQUtils.AmountsArithmetic.toNumber(ModelUtils.get(balances, i, "balance"), decimals)
+            }
+            return totalBalance
+        }
+    }
+
+    property SortFilterProxyModel customSFPM: SortFilterProxyModel {
+        sourceModel: d.assetsWithFilteredBalances
         proxyRoles: [
-            ExpressionRole {
-                name: "currentBalance"
-                expression: model.enabledNetworkBalance.amount
+                FastExpressionRole {
+                id: filter
+                 name: "currentBalance"
+                 expression: d.getTotalBalance(model.balances, model.decimals, root.addressFilters, root.networkFilters)
+                 expectedRoles: ["balances", "decimals"]
+            },
+            FastExpressionRole {
+                 name: "currentCurrencyBalance"
+                 expression: d.getTotalBalance(model.balances, model.decimals, root.addressFilters, root.networkFilters) *
+                             model.marketDetails.currencyPrice.amount
+                 expectedRoles: ["balances", "marketDetails", "decimals"]
             },
-            ExpressionRole {
-                name: "currentCurrencyBalance"
-                expression: model.enabledNetworkCurrencyBalance.amount
+            FastExpressionRole {
+                 name: "tokenPrice"
+                 expression: model.marketDetails.currencyPrice.amount
+                 expectedRoles: ["marketDetails"]
             },
-            ExpressionRole {
-                name: "tokenPrice"
-                expression: model.currencyPrice.amount
+            FastExpressionRole {
+                 name: "isCommunityAsset"
+                 expression: !!model.communityId
+                 expectedRoles: ["communityId"]
             }
         ]
         filters: [
             ExpressionFilter {
                 expression: {
-                    d.controller.settingsDirty
-                    return d.tokenIsVisible(model.symbol, model.currentCurrencyBalance) && (customFilter.isCommunity ? !!model.communityId : !model.communityId)
+                    if (!!model.communityId) {
```

Comment:
No, we want to filter also non-community tokens

Reply:
The original expression:
```js
(customFilter.isCommunity ? !!model.communityId : !model.communityId)
```
is different

Reply:
yes I am aware, you needed that for the 2 list views but now that we have a single one, 
`(customFilter.isCommunity ? !!model.communityId : !model.communityId)` is not needed.
The reason for having condition for the community is that we check for currency balance, but currency balance for community tokens will always be 0 as they have no real price on any market, hence they should not be filtered out by this filter.

Reply:
No, that's not the point; this function `d.tokenIsVisible()` also checks for those explicitly hidden by the user (in Manage tokens). Now you're letting those slip

Reply:
gotcha! Ive kept the filter for community tokens too now :)

Reply:
@caybro Ive fixed all your comments now, please check when you can :)